File: python/tmps/tmp1/main.py

def main(): 
    nums: list = [1, 2, 3, 3]
    result: bool = run(nums)

    print(f"Result: {result}")

def run(nums: list[int]) -> bool:
    seen: set[int] = set()

    for val in nums:
        if val in seen:
            return True
        seen.add(val)

    return False


if __name__ == "__main__":
    main()

File: cpp/tmps/tmp1/main.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

std::pair<int, int> run(const std::vector<int>& nums, int target) {
    std::unordered_map<int, int> seen;

    for (int i = 0; i < nums.size(); i++) {
        int num = nums[i];
        int complement = target - num;

        if (seen.find(complement) != seen.end()) {
            int compIdx = seen[complement];
            return {compIdx, i};
        }

        seen[num] = i;
    }

    return {-1, -1};
}

int main() {
    std::vector<int> nums = {1, 2, 3, 4};
    std::vector<int> *pNums = &nums;
    int target = 7;

    std::pair<int, int> result = run(*pNums, target);

    if (result.first != -1) {
        std::cout << result.first << result.second << "\n";
    } else {
        std::cout << false << "\n";
    }
}
